intelmq/bots/experts/idea/expert.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+from collections import Sequence, Mapping
+from base64 import b64decode
+from uuid import uuid4
+from urllib.parse import quote_plus
+
+from intelmq.lib.bot import Bot
+
+
+def quot(s):
+    return quote_plus(s or "")
+
+
+def addr4(s):
```

Comment:
Note that this code does no validation, other parts of the code are resonsible of this. These small helper function only have the task to differentiate between the two version.